src/array/ops/utf8.rs

```diff
@@ -17,6 +17,19 @@ impl Utf8Array {
         self.binary_broadcasted_compare(pattern, |data: &str, pat: &str| data.contains(pat))
     }
 
+    pub fn length(&self) -> DaftResult<UInt64Array> {
+        let self_arrow = self.downcast();
+        let arrow_result = self_arrow
+            .iter()
+            .map(|val| {
+                let v = val?;
+                Some(v.len() as u64)
+            })
+            .collect::<arrow2::array::UInt64Array>()
+            .with_validity(self_arrow.validity().cloned());
+        Ok(UInt64Array::from((self.name(), Box::new(arrow_result))))
```

Comment:
I initially opted for using the `UInt64Array::from()` path for construction for the sake of conciseness, but if the `Box` --> `Arc` path isn't as cheap as we thought, it probably makes sense to port this to a direct `DataArray::<UInt64Type>::new()` construction with the `Arc` pointers. I'll make that change!